src/sql/src/plan/statement.rs

```diff
@@ -446,13 +447,30 @@ fn handle_show_databases(
     scx: &StatementContext,
     ShowDatabasesStatement { filter }: ShowDatabasesStatement,
 ) -> Result<Plan, anyhow::Error> {
-    let rows = scx
-        .catalog
-        .list_databases()
-        .map(|database| vec![Datum::from(database)])
-        .collect();
-
-    finish_show_where(scx, filter, rows, &SHOW_DATABASES_DESC)
+    let select = Select::default()
+        .from(TableWithJoins {
+            relation: TableFactor::Table {
+                name: ObjectName(vec![Ident::new("mz_databases")]),
+                alias: None,
+            },
+            joins: vec![],
+        })
+        .project(SelectItem::Expr {
+            expr: Expr::Identifier(vec![Ident::new("database")]),
+            alias: None,
+        })
+        .selection(filter);
+    handle_select(
+        scx,
+        SelectStatement {
+            query: Box::new(Query::select(select)),
+            as_of: None,
+        },
+        &Params {
+            datums: Row::pack(&[]),
+            types: vec![],
+        },
+    )
```

Comment:
Brilliant! You'll probably find as you continue to port things over that the guts of this function get repeated a bunch—up to you to figure out if you want to share some code + how. (But I think spelling it all out explicitly like this while there is only one example is 100% the way to go.)